GPTPullRequestReview/src/git.ts

```diff
@@ -15,11 +15,13 @@ export async function getChangedFiles(targetBranch: string) {
   await git.addConfig('core.quotepath', 'false');
   await git.fetch();
 
-  const diffs = await git.diff([targetBranch, '--name-only', '--diff-filter=AM']);
+  const diffs = await git.diff([targetBranch, '--name-only', '--diff-filter=AMD']);
```

Comment:
here we already filter the changed files and we exclude deleted files, why include again the deleted files filter again in line 22 ?

Reply:
Yes, you are correct. Sorry, didn't really think that one through. :)
Was just trying (a little bit to quick) to fix this issue I experience related to files deleted in the PR that seems to be handled even though they should not:

2023-09-12T14:05:01.3999340Z Start reviewing tools/ServiceHelper/Logging/SerilogHelper.cs with openai ...
2023-09-12T14:05:01.4000070Z Start reviewing tools/ServiceTestHelper/Wiring/ServiceTestFactory.cs with openai ...
2023-09-12T14:05:01.4281262Z ##[error]Unhandled: fatal: ambiguous argument '.azure-pipelines/templates/deploy.yml': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

2023-09-12T14:05:01.4297670Z ##[error]Error: fatal: ambiguous argument '.azure-pipelines/templates/deploy.yml': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

    at Object.action (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1274:25)
    at PluginStore.exec (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1309:29)
    at Promise (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1676:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1674:16)
    at GitExecutorChain.<anonymous> (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1658:44)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:55:24)
2023-09-12T14:05:01.4559300Z ##[error]Unhandled: fatal: ambiguous argument '.azure-pipelines/templates/infra.yml': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

2023-09-12T14:05:01.4564525Z ##[error]Error: fatal: ambiguous argument '.azure-pipelines/templates/infra.yml': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

    at Object.action (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1274:25)
    at PluginStore.exec (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1309:29)
    at Promise (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1676:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1674:16)
    at GitExecutorChain.<anonymous> (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:1658:44)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/GPTPRReviewer_54bfc437-996c-4844-9f68-cd33e7a56682/0.6.58/node_modules/simple-git/dist/cjs/index.js:55:24)
2023-09-12T14:05:01.7070228Z api_key_source: openai
2023-09-12T14:05:01.7070652Z endpoint: https://api.openai.com/v1/chat/completions
2023-09-12T14:05:01.7070999Z AuthType: Authorization


Where deploy.yml and infra.yml have been deleted in the PR but seems to be passed through the filter and therefore is causing errors and making the task to fail.